src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -148,6 +164,92 @@ public function update(
         }
     }
 
+    /**
+     * Synchronizes modified products from the cart with the order
+     *
+     * @param CartProductUpdate[] $modifiedProducts
+     * @param Cart $cart
+     * @param Order $order
+     *
+     * @throws OrderException
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    private function updateOrderModifiedProducts(array $modifiedProducts, Cart $cart, Order $order): void
+    {
+        $productsToAddToOrder = [];
+        foreach ($modifiedProducts as $modifiedProduct) {
+            if (null === $this->findProductInCart($modifiedProduct, $cart)) {
```

Comment:
In this foreach, probably you will call `findProductInCart` and `findProductInOrder` multiple times. Why not calling them just once and set them into variables?